### PR TITLE
refactorings as per plenum

### PR DESCRIPTION
**Pull request by lovesh**
Signed-off-by: Lovesh <lovesh.harchandani@evernym.com>

### CODE REVIEW

File: indy_node/server/client_authn.py [hunk 1]
```diff
@@ -1,35 +1,41 @@
...
 
-class TxnBasedAuthNr(NaclAuthNr):
+class LedgerBasedAuthNr(CoreAuthMixin, NaclAuthNr):
     """
     Transaction-based client authenticator.
     """
 
+    write_types = CoreAuthMixin.write_types.union({ATTRIB, SCHEMA, CLAIM_DEF,
+                                                   POOL_CONFIG, POOL_UPGRADE})
+    query_types = CoreAuthMixin.query_types.union({GET_NYM, GET_ATTR, GET_SCHEMA,
```
Comment (ashcherbakov) [inline indy_node/server/client_authn.py]: I think we already have a list with query txns: `openTxns` in `constants/py`. Let's use just one list in all places.

File: indy_node/server/node.py [hunk 2]
```diff
@@ -451,12 +358,14 @@ def executeDomainTxns(self, ppTime, reqs: List[Request], stateRoot,
         :param ppTime: the time at which PRE-PREPARE was sent
         :param req: the client REQUEST
         """
-        return self.commitAndSendReplies(self.reqHandler, ppTime, reqs,
+        req_handler = self.get_req_handler(DOMAIN_LEDGER_ID)
+        return self.commitAndSendReplies(req_handler, ppTime, reqs,
                                          stateRoot, txnRoot)
 
     def executeConfigTxns(self, ppTime, reqs: List[Request], stateRoot,
```
Comment (ashcherbakov) [inline indy_node/server/node.py]: Can we use generic code for excuteXTxns?

File: indy_node/server/node.py [hunk 3]
```diff
@@ -319,52 +318,8 @@ def authNr(self, req):
         else:
             return super().authNr(req)
 
-    def isSignatureVerificationNeeded(self, msg: Any):
```
Comment (ashcherbakov) [inline indy_node/server/node.py]: We need to remove it in plenum too, and make request handlers decide whether signature verification is needed.

Comment (lovesh) [reply indy_node/server/node.py]: `isSignatureVerificationNeeded` is intended for general query txns, like GET_TXN and GET_TXNS, i will change `openTxns`

File: indy_node/server/node.py (hunk 2 above)
Comment (lovesh) [reply indy_node/server/node.py]: I used more generic code in the next PR

File: indy_node/persistence/idr_cache.py [hunk 4]
```diff
@@ -79,55 +80,80 @@ def unpackIdrValue(value):
...
+        try:
+            assert super().first_batch_idr == stateRoot, 'The first created ' \
+                                                         'batch has not been ' \
+                                                         'committed or reverted ' \
+                                                         'and yet another batch ' \
+                                                         'is trying to be ' \
+                                                         'committed, {} {}'.\
+                format(self.unCommitted[0][0], stateRoot)
+            super().commit_batch()
+        except Exception:
```
[the remark this answers is not in the document]
Comment (lovesh) [reply indy_node/persistence/idr_cache.py]: Done

File: indy_node/persistence/idr_cache.py [hunk 5]
```diff
@@ -79,55 +80,80 @@ def unpackIdrValue(value):
...
-            'The first created batch has ' \
-            'not been committed or ' \
-            'reverted and yet another ' \
-            'batch is trying to be ' \
-            'committed, {} {}'\
-            .format(self.unCommitted[0][0], stateRoot)
-        self._keyValueStorage.setBatch([(idr, val) for idr, val in
-                                        self.unCommitted[0][1].items()])
-        self.unCommitted = self.unCommitted[1:]
+        # if self.unCommitted:
```
[the remark this answers is not in the document]
Comment (lovesh) [reply indy_node/persistence/idr_cache.py]: Done

File: indy_node/server/client_authn.py (hunk 1 above)
Comment (lovesh) [reply indy_node/server/client_authn.py]: `openTxns` is meant for txns like `GET_TXNS`, i have made that change

File: indy_node/persistence/idr_cache.py [hunk 6]
```diff
@@ -79,55 +80,80 @@ def unpackIdrValue(value):
...
-            # iterating over `currentBatchOps and unCommitted`
-            # in reverse to get the latest value
-            for key, cache in reversed(self.currentBatchOps):
-                if key == idr:
-                    return self.unpackIdrValue(cache)
-            for _, cache in reversed(self.unCommitted):
-                if encoded_idr in cache:
-                    return self.unpackIdrValue(cache[encoded_idr])
-        value = self._keyValueStorage.get(encoded_idr)
+        idr = idr.encode()
```
[the remark this answers is not in the document]
Comment (lovesh) [reply indy_node/persistence/idr_cache.py]: Done

File: indy_node/persistence/idr_cache.py [hunk 7]
```diff
@@ -79,55 +80,80 @@ def unpackIdrValue(value):
...
-    def set(self, idr, seqNo, txnTime, ta=None, role=None, verkey=None, isCommitted=True):
+    def set(self, idr, seqNo, txnTime,
+            ta=None, role=None, verkey=None, isCommitted=True):
+        idr = idr.encode()
         val = self.packIdrValue(seqNo, txnTime, ta, role, verkey)
-        if isCommitted:
-            self._keyValueStorage.put(idr, val)
-        else:
-            self.currentBatchOps.append((idr, val))
+        # if isCommitted:
```
[the remark this answers is not in the document]
Comment (lovesh) [reply indy_node/persistence/idr_cache.py]: Done

File: indy_node/persistence/idr_cache.py [hunk 8]
```diff
@@ -79,55 +80,80 @@ def unpackIdrValue(value):
...
+        #     self.currentBatchOps.append((idr, val))
+        super().set(idr, val, is_committed=isCommitted)
 
     def close(self):
         self._keyValueStorage.close()
 
     def currentBatchCreated(self, stateRoot):
-        self.unCommitted.append((stateRoot, OrderedDict(self.currentBatchOps)))
-        self.currentBatchOps = []
+        # self.unCommitted.append((stateRoot, OrderedDict(self.currentBatchOps)))
```
[the remark this answers is not in the document]
Comment (lovesh) [reply indy_node/persistence/idr_cache.py]: Done

File: indy_node/server/client_authn.py (hunk 1 above)
Comment (ashcherbakov) [reply indy_node/server/client_authn.py]: Why can't we move openTxns here as well? so, we have all logic related to signature verification skipping in one place?

Comment (ashcherbakov) [inline indy_node/server/client_authn.py]: Can we get `query_types` from registered req handlers? Looks like we have a duplication here (query_types in both Requesthandlers and LedgerBasedAuthNr). Moreover, if new req handlers are added, then we need to change `query_types` in LedgerBasedAuthNr manually.

Comment (lovesh) [reply indy_node/server/client_authn.py]: Ok

**Pull request merged**